### PR TITLE
fix: Add POM configuration to allow for maven deploy phase

### PR DESCRIPTION
**Pull request by larshelge**
* Adds the necessary configuration to `pom.xml` to allow for deploying this project/artefact to the Sonatype Maven repository. Adds scm, issue management, distribution management. Adds required plugins for javadoc, source, nexus staging and gpg key generation.

* The gpg library is now required to be installed locally.

* Sets version to 0.1.0.

* The 0.1.0 version is now deployed to Sonatype, which syncs with Maven central.

* A new version can now be deployed with `mvn deploy`.

### CODE REVIEW

**Review by jbee — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -94,6 +118,58 @@
                     <lineEndings>UNIX</lineEndings>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-source-plugin</artifactId>
+                <version>3.0.1</version>
```
Comment (teleivo) [inline pom.xml:124]: can you please update it to the latest version: 3.2.0

File: pom.xml [hunk 2]
```diff
@@ -94,6 +118,58 @@
...
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.0.0</version>
```
Comment (teleivo) [inline pom.xml:137]: please update to the latest version: 3.3.1

File: pom.xml [hunk 3]
```diff
@@ -94,6 +118,58 @@
...
+                    <serverId>ossrh</serverId>
+                    <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+                    <autoReleaseAfterClose>true</autoReleaseAfterClose>
+                    <stagingProgressTimeoutMinutes>15</stagingProgressTimeoutMinutes>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-gpg-plugin</artifactId>
+                <version>1.6</version>
```
Comment (teleivo) [inline pom.xml:162]: latest version: 3.0.1

**Review by teleivo — COMMENTED** [review]
please update to the latest plugin versions 😄 
I had prepared https://github.com/dhis2/json-tree/pull/3 
https://github.com/actions/setup-java/blob/ae26cabe43aaa84f8539613cdb171b33610fa704/docs/advanced-usage.md#publishing-using-apache-maven

which relies on the release plugin which is also able to publish to a staging repo
https://maven.apache.org/maven-release/maven-release-plugin/stage-mojo.html

not sure if this can completely replace nexus-staging-maven-plugin

I haven't merged my PR because I am missing a GPG key. Interpreting the discussion we had on Slack I guess the only option I see right now is generating yet another GPG key, putting it in the repo secrets and using it to sign the jar.

**Comment by larshelge** [comment]
@teleivo I have sent you the Maven repo credentials separately. I have upgraded the plugin versions you mentioned.

Regarding using the release plugin - I am sure there are several ways to deploy. Though we already have several (four or so) projects which we deploy to our Sonatype repo using this approach, so I am thinking it adds value to keep this consistent too.

**Comment by teleivo** [comment]
> @teleivo I have sent you the Maven repo credentials separately. I have upgraded the plugin versions you mentioned.
> 
> Regarding using the release plugin - I am sure there are several ways to deploy. Though we already have several (four or so) projects which we deploy to our Sonatype repo using this approach, so I am thinking it adds value to keep this consistent too.

We already had the Maven repo credentials, thank you 😄 Austin put them in the GitHub org secrets so they are accessible in actions via ${{ secrets.OSSRH_USERNAME }} and ${{ secrets.OSSRH_PASSWORD }} (which I refer to https://github.com/dhis2/json-tree/pull/3)

What we are missing is a gpg key to sign the artifact. That is also the reason why this PRs build fails. The DHIS2 repos that sign/publish that I've found seem to each use their own (created by a developer I guess) gpg key. I've asked in the devops/security channel what the best practice would be. I think a developer independent one, put in the GitHub org secrets would be good. I don't think anyone is working on it though. So we need a volunteer to tackle that or a volunteer putting a gpg key into this repos secrets so this PR build fine and we can publish 😄

**Comment by larshelge** [comment]
Okay I agree. We could have a shared secret key. One question is whether we should integrate this into the CI pipeline somehow then, so that we don't have to distribute and install this key on many local machines?

**Comment by teleivo** [comment]
> 

Yes, we could add the GPG key as a GitHub org secret. This way any DHIS2 repo would be able to use it without needing it locally. A GitHub action would then look like https://github.com/dhis2/json-tree/pull/3

If by `integrate into the CI pipeline` you also mean how the GPG key gets int the GitHub secrets. So gpg key generation/rotation then that would be great too. Andreas mentioned the use of AWS key management service. I am unsure how much work that would be though. GitHub has at least an API for their secrets https://docs.github.com/en/rest/reference/actions#secrets Devops team is probably best suited to answer that 😄

**Pull request merged**